### PR TITLE
chore: lint require i18n-resources first in status-im2.core

### PR DESCRIPTION
**Pull request by yqrashawn**
### Summary
add linter, fail when `status-im2.setup.i18n-resources` is not the first namespace in  `status-im2.core` ns form

detail in this comment https://github.com/status-im/status-mobile/pull/17618#discussion_r1361275489

error looks like

CleanShot 2023-11-15 at 12 52 52@2x

### Testing notes
dev exp only PR

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

File: scripts/lint-require-i18n-resource-first.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+
+if rg --quiet --multiline '^\(ns.*\n^\s*\(:require\n(^\s*(;|#_).*\n)*(^\s*\[status-im2\.setup\.i18n-resources\W)' src/status_im2/core.cljs; then
```
Comment (jakubgs) [inline scripts/lint-require-i18n-resource-first.sh]: Where exactly are you getting this `rg` from?

Comment (jakubgs) [reply scripts/lint-require-i18n-resource-first.sh]: Oh, I see, it was added here:
https://github.com/status-im/status-mobile/blob/45df308dd0d0e2b3945a70f514dcf5fc5f596abe/nix/shells.nix#L23-L24

**Review by jakubgs — APPROVED** [review]
We are creating quite a collection of `lint` scripts. Maybe we should just have a `scripts/lint` folder?

**Review by ilmotta — APPROVED** [review]

**Comment by yqrashawn** [comment]
added `lint` folder, renamed `lint-xxx` scripts to `xxx`

**Pull request merged**